File: typescript/agent.ts
/**
 * Report API agent — TypeScript example.
 *
 * Generates financial reports and resumes with result.
 *
 * Usage:
 *   export AXME_API_KEY="<agent-key>"
 *   npx tsx agent.ts
 */

import { AxmeClient } from "@axme/axme";

const AGENT_ADDRESS = "report-api-demo";

async function handleIntent(client: AxmeClient, intentId: string) {
  const intentData = await client.getIntent(intentId);
  const intent = intentData.intent ?? intentData;
  let payload = intent.payload ?? {};
  if (payload.parent_payload) {
    payload = payload.parent_payload;
  }

  const reportType = payload.report_type ?? "unknown";
  const fmt = payload.format ?? "pdf";
  const year = payload.fiscal_year ?? "2025";

  console.log(`  Generating ${reportType} report (${fmt}) for FY${year}...`);
  await new Promise((r) => setTimeout(r, 2000));

  const result = {
    action: "complete",
    report_url: `https://reports.example.com/FY${year}-${reportType}.${fmt}`,
    pages: 128,
    generated_at: new Date().toISOString(),
  };

  await client.resumeIntent(intentId, result, { ownerAgent: "report-api-demo" });
  console.log(`  Report ready: ${result.report_url}`);
}

async function main() {
  const apiKey = process.env.AXME_API_KEY;
  if (!apiKey) {
    console.error("Error: AXME_API_KEY not set.");
    process.exit(1);
  }

  const client = new AxmeClient({ apiKey });

  console.log(`Agent listening on ${AGENT_ADDRESS}...`);
  console.log("Waiting for intents (Ctrl+C to stop)\n");

  for await (const delivery of client.listen(AGENT_ADDRESS)) {
    const intentId = delivery.intent_id;
    const status = delivery.status;
    if (intentId && ["DELIVERED", "CREATED", "IN_PROGRESS"].includes(status)) {
      console.log(`[${status}] Intent received: ${intentId}`);
      try {
        await handleIntent(client, intentId);
      } catch (e) {
        console.error(`  Error: ${e}`);
      }
    }
  }
}

main().catch(console.error);

File: java/Agent.java
/*
 * Report API agent — Java example.
 *
 * Fetches an intent by ID, generates a financial report, and resumes with result.
 *
 * Usage:
 *   export AXME_API_KEY="<agent-key>"
 *   javac -cp axme-sdk.jar Agent.java
 *   java -cp .:axme-sdk.jar Agent <intent_id>
 */

import dev.axme.sdk.AxmeClient;
import dev.axme.sdk.AxmeClientConfig;
import dev.axme.sdk.RequestOptions;
import java.time.Instant;
import java.util.Map;

public class Agent {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) {
            System.err.println("Usage: java Agent <intent_id>");
            System.exit(1);
        }

        String apiKey = System.getenv("AXME_API_KEY");
        if (apiKey == null || apiKey.isEmpty()) {
            System.err.println("Error: AXME_API_KEY not set.");
            System.exit(1);
        }

        String intentId = args[0];
        var client = new AxmeClient(AxmeClientConfig.forCloud(apiKey));

        System.out.println("Processing intent: " + intentId);

        var intentData = client.getIntent(intentId, new RequestOptions());
        @SuppressWarnings("unchecked")
        var intent = (Map<String, Object>) intentData.getOrDefault("intent", intentData);
        @SuppressWarnings("unchecked")
        var payload = (Map<String, Object>) intent.getOrDefault("payload", Map.of());
        if (payload.containsKey("parent_payload")) {
            @SuppressWarnings("unchecked")
            var pp = (Map<String, Object>) payload.get("parent_payload");
            payload = pp;
        }

        String reportType = (String) payload.getOrDefault("report_type", "unknown");
        String fmt = (String) payload.getOrDefault("format", "pdf");
        String year = (String) payload.getOrDefault("fiscal_year", "2025");

        System.out.println("  Generating " + reportType + " report (" + fmt + ") for FY" + year + "...");
        Thread.sleep(2000);

        String reportUrl = "https://reports.example.com/FY" + year + "-" + reportType + "." + fmt;
        var result = Map.<String, Object>of(
            "action", "complete",
            "report_url", reportUrl,
            "pages", 128,
            "generated_at", Instant.now().toString()
        );

        client.resumeIntent(intentId, result, new RequestOptions());
        System.out.println("  Report ready: " + reportUrl);
    }
}
